FirebaseDatabase/Sources/Utilities/FNextPushId.m

```diff
@@ -59,4 +65,43 @@ + (NSString *)get:(NSTimeInterval)currentTime {
     return [NSString stringWithString:id];
 }
 
++ (NSString *)nextAfter:(NSString *_Nonnull)key {
+    NSInteger keyAsInt;
+    if ([FUtilities tryParseStringToInt:key asInt:&keyAsInt]) {
+        return [FUtilities
+            ieee754StringForNumber:[NSNumber numberWithInteger:(keyAsInt + 1)]];
+    }
+    NSString *charFormat = @"%C";
+    NSMutableString *next = [NSMutableString stringWithString:key];
+    if ([next length] < MAX_KEY_LEN) {
+        [next insertString:MIN_PUSH_CHAR atIndex:[key length]];
+        return next;
+    }
+    unsigned long i = [next length] - 1;
+
+    while (i >= 0 && [[next substringWithRange:NSMakeRange(i, i + 1)]
```

Comment:
Can you use https://developer.apple.com/documentation/foundation/nsstring/1414645-characteratindex?language=objc ?